relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py

```diff
@@ -145,16 +145,29 @@ def __call__(self, r):
         r.register_hook('response', self.handle_407)
         return r
 
-def getRequest(url, stream):
+def getRequest(url, stream, timeout = 60):
   if (PROXY_URL == ""):
-    return requests.get(url, auth=(USERNAME, PASSWORD), stream=stream)
+    return requests.get(url, auth=(USERNAME, PASSWORD), stream=stream, timeout = timeout)
   else:
     proxies = { "https": PROXY_URL, "http": PROXY_URL }
     if (PROXY_USERNAME != "" and PROXY_PASSWORD != "" ):
       auth = HTTPProxyDigestAuth(PROXY_USERNAME, PROXY_PASSWORD)
-      return requests.get(url, proxies = proxies, auth = auth, stream=stream)
+      return requests.get(url, proxies = proxies, auth = auth, stream=stream, timeout = timeout)
     else:
-      return requests.get(url, auth=(USERNAME, PASSWORD), proxies = proxies, stream=stream)
+      return requests.get(url, auth=(USERNAME, PASSWORD), proxies = proxies, stream=stream, timeout = timeout)
+
+# Check if plugin URL defined in configuration file can be accessed, so we can fail early our cmmands
+def check_url():
+  try:
+    r = getRequest(URL, False, 5)
+    if r.status_code == 401:
+      fail("Received a HTTP 401 Unauthorized error when trying to get %s. Please check your credentials in %s"%(completeUrl, CONFIG_PATH))
+    elif r.status_code > 400:
+      fail("Received a HTTP %s error when trying to get %s"%(r.status_code, completeUrl))
+    else:
+      True
+  except Exception as e:
+    fail("An error occured while checking access to %s:\n%s"%(URL, e))
```

Comment:
This will make the command fail on systems not connected to the internet. We either need to return 0 in this case or (maybe preferably) to add || true to https://github.com/Normation/rudder-packages/pull/2368/files. But we shouldn't display an error anyway as it is an expected behavior.